index.js

```diff
@@ -144,6 +149,30 @@ class HLSSpliceVod {
     });
   }
 
+  insertBumper(bumperMasterManifestUri, _injectBumperMasterManifest, _injectBumperMediaManifest) {
+    return new Promise((resolve, reject) => {
+      this._parseAdMasterManifest(bumperMasterManifestUri, _injectBumperMasterManifest, _injectBumperMediaManifest)
+      .then(bumper => {
+        const bandwidths = Object.keys(this.playlists);
+        for (let b = 0; b < bandwidths.length; b++) {
+          const bw = bandwidths[b];
+
+          const bumperPlaylist = bumper.playlist[findNearestBw(bw, Object.keys(bumper.playlist))];
+          const bumperLength = bumperPlaylist.items.PlaylistItem.length;
+          this.bumperDuration = 0;
+          for (let j = 0; j < bumperLength; j++) {
+            this.playlists[bw].items.PlaylistItem.splice(j, 0, bumperPlaylist.items.PlaylistItem[j]);
+            this.bumperDuration += (bumperPlaylist.items.PlaylistItem[j].get('duration') * 1000);
+          }
+          this.playlists[bw].items.PlaylistItem[bumperLength].set('discontinuity', true);
+          this.playlists[bw].set('targetDuration', this.targetDuration);
+          this.bumperOffset = bumperLength;
```

Comment:
It all looks good and seems to work perfectly. But bumperOffset doesn't look like it's ever used in the insertAdAt( ) function.

Reply:
You are correct. That line should be removed.